src/lib/LibBurn.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+/// @title LibBurn
+/// @notice Utilities for burning Gas and Ether
+library LibBurn {
+    /// Burns a given amount of ETH.
+    /// @param _amount Amount of ETH to burn.
+    function eth(uint256 _amount) internal {
+        new Burner{ value: _amount }();
+    }
```

Comment:
Great idea to get rid of the `SELFDESTRUCT` op here! What do you think about something like this rather than a Sol contract?

```solidity
    /// Burns a given amount of ETH.
    /// @param _amount Amount of ETH to burn.
    function eth(uint256 _amount) internal {
        assembly ("memory-safe") {
            // Store creation code of no-op contract
            // PUSH1 0xFE (Invalid opcode)
            // PUSH1 0x00  \
            // MSTORE       > Replace with PUSH0 / CHAINID for mainnet.
            // PUSH1 0x01  /
            // PUSH1 0x1F
            // RETURN
            mstore(0x00, 0x60fe6000526001601ff3)
            let success := create(_amount, 0x16, 0x0a)

            if iszero(success) {
                // "BurnFailed()" error
                mstore(0x00, 0x6f16aafc)
                revert(0x1c, 0x04)
            }
        }
    }
```

Reply:
Oh good point, this is a much much better way to burn eth rather than using the kind of ugly solidity Syntax too